qa/rpc-tests/key_import_export.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python2
+# Copyright (c) 2017 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from time import sleep
+from decimal import Decimal
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, assert_greater_than, sync_blocks
+
+import logging
+
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+
+class KeyImportExportTest (BitcoinTestFramework):
+
+    def run_test(self):
+        [alice, bob, charlie, miner] = self.nodes
+
+        def wait_until_miner_sees(txid):
+            attempts = 123
+            delay = 0.5
+
+            for _ in range(attempts):
+                try:
+                    miner.getrawtransaction(txid)
+                except Exception:
+                    logging.debug(
+                        'txid %r not yet seen by miner; sleep %r',
+                        txid,
+                        delay,
+                    )
+                    sleep(delay)
+                else:
+                    return
+
+            raise Exception(
+                'miner failed to see txid {!r} after {!r} attempts...'.format(
+                    txid,
+                    attempts,
+                ),
+            )
+
+        def alice_to_bob(amount):
+            txid = alice.sendtoaddress(addr, Decimal(amount))
+            wait_until_miner_sees(txid)
+            miner.generate(1)
+            sync_blocks(self.nodes)
+
+        def verify_utxos(node, amounts):
+            utxos = node.listunspent(1, 10**9, [addr])
+
+            def cmp_confirmations_high_to_low(a, b):
+                return cmp(b["confirmations"], a["confirmations"])
+
+            utxos.sort(cmp_confirmations_high_to_low)
+
+            try:
+                assert_equal(amounts, [utxo["amount"] for utxo in utxos])
+            except AssertionError:
+                logging.error(
+                    'Expected amounts: %r; utxos: %r',
+                    amounts, utxos)
+                raise
+
+        # The first address already mutated by the startup process, ignore it:
+        # BUG: Why is this necessary?
+        bob.getnewaddress()
```

Comment:
This is really ugly and shouldn't be necessary. I guess it's okay for now.

Reply:
The fix is to create a clean chain. I pushed a commit.